android/src/main/java/com/applovin/reactnative/AppLovinMAXModule.java

```diff
@@ -1897,6 +1903,39 @@ private WritableMap createNetworkResponseInfo(final MaxNetworkResponseInfo respo
         return networkResponseObject;
     }
 
+    // Lifecycle Events
+
+    @Override
+    public void onHostResume() { }
+
+    @Override
+    public void onHostPause() { }
+
+    @Override
+    public void onHostDestroy()
+    {
+        // Converts mAdViews to a list since destroyAdView will modify mAdViews
+        List<MaxAdView> adViews = new ArrayList<MaxAdView>( mAdViews.values() );
+        for ( Iterator<MaxAdView> iterator = adViews.iterator(); iterator.hasNext(); )
+        {
+            MaxAdView adView = iterator.next();
+            iterator.remove();
```

Comment:
Ah, my bad as well. Didn't notice that we were making a copy. In that case, we don't need to remove the item from the copy or even use an iterator.